### DISCUSSION TITLE
Unlocked GPU Benefits

### DISCUSSION THREAD

**Discussion by ghost** [rg35xx-cfw/Koriki#36] [Q&A]
I couldn't find the answer anywhere - does unlocking GPU in these firmwares, apart from making some more advanced consoles possible to emulate, improve performance of any of the systems previosuly available on both stock and Garlic OS (or basically non-3D systems, as I guess PS1 will benefit too), thus maybe making the battery life less miserable?

**Comment by JohnnyonFlame** [comment]
You get shaders and ports that might require said GPU, such as Maldita Castilla, AM2R, Spelunky, etc. Also a few systems such as N64 and PSP, but those are hardly great experiences here.

**Comment by ghost** [comment]
Yeah, but as I said I'm more interested whether it has impact on performance of things that worked before. It's weird that no one knows whether there is any 2D acceleration-sort-of-thing or not.

**Reply by junkaddr** [reply]
Why don't you test it and find out?

**Reply by ghost** [reply]
That's very helpful, thanks 🙄

**Reply by acmeplus** [reply]
It's not completely straightforward to compare both. GarlicOS has a highly optimized framebuffer implementation that manages to run most of the 2D at the maximum possible speed of the devices. On the other hand the GPU implementation is not very optimized for this device specifically (other than some great SDL2 changes done by @JohnnyonFlame) and the most part uses standard SDL2 code. The GPU performs all the scaling, rendering, etc. so while those need to be performed manually in the framebuffer implementation (using things like @eggs neon scalers) the GPU basically does all those automatically.

If you are asking about specific benchmarks, no, there are none that I'm aware of.

**Comment by ghost** [comment]
No no, no benchmarks - rather a general gist of the difference :) thanks acmeplus, that makes it clearer
